[Test framework] Embedded server -> dependency download error when no version is specified

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

testsuite

### Describe the bug

there is a bug, when loading e.g. jdbc driver for embedded server: https://github.com/keycloak/keycloak/blob/main/test-framework/db-oracle/src/main/java/org/keycloak/testframework/database/OracleDatabaseSupplier.java#L22

The problem is caused by missing version when downloading a dependency. It should be added then as a provider to Keycloak in an embedded mode.

```
2025-04-15 16:18:27,285 ERROR [io.quarkus.bootstrap.resolver.maven.NonBlockingModelResolutionTaskRunner] (main) 1): io.quarkus.bootstrap.resolver.maven.DeploymentInjectionException: Failed to resolve artifact com.oracle.database.jdbc:ojdbc17:jar:
	at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver.resolve(ApplicationDependencyResolver.java:869)
	at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDep.getResolvedArtifact(ApplicationDependencyResolver.java:714)
	at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDep.visitRuntimeDependency(ApplicationDependencyResolver.java:557)
	at io.quarkus.bootstrap.resolver.maven.NonBlockingModelResolutionTaskRunner.lambda$run$0(NonBlockingModelResolutionTaskRunner.java:34)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run$$$capture(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
	at java.base/java.util.concurrent.ForkJoinTask.doExec$$$capture(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: com.oracle.database.jdbc:ojdbc17:jar: (absent): Could not find artifact com.oracle.database.jdbc:ojdbc17:jar: in central (https://repo.maven.apache.org/maven2)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:473)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:261)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:243)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact(DefaultRepositorySystem.java:278)
	at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver.resolve(ApplicationDependencyResolver.java:863)
	... 12 more
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact com.oracle.database.jdbc:ojdbc17:jar: in central (https://repo.maven.apache.org/maven2)
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:42)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:417)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:260)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:537)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:449)
	... 16 more
```

### Version

nightly

### Regression

- [ ] The issue is a regression

### Expected behavior

driver should be loaded

### Actual behavior

embedded server is failing on maven resolver

### How to Reproduce?

run a random test from the newly migrated package tests/base and set KC_TEST_DATABASE=oracle

### Anything else?

_No response_

--- a/quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/Maven.java
+++ b/quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/Maven.java
@@ -46,6 +46,10 @@ import org.eclipse.aether.resolution.DependencyResolutionException;
 public final class Maven {
 
     public static Path resolveArtifact(String groupId, String artifactId) {
+        return getArtifact(groupId, artifactId).getFile().toPath();
+    }
+
+    public static Artifact getArtifact(String groupId, String artifactId) {
         try {
             BootstrapMavenContext ctx = bootstrapCurrentMavenContext();
             LocalProject project = ctx.getCurrentProject();
@@ -72,7 +76,7 @@ public final class Maven {
                             ctx.getRepositorySystemSession(),
                             new ArtifactRequest().setArtifact(artifact)
                                     .setRepositories(remoteRepositories))
-                    .getArtifact().getFile().toPath();
+                    .getArtifact();
         } catch (Exception cause) {
             throw new RuntimeException("Failed to resolve artifact: " + groupId + ":" + artifactId, cause);
         }

--- a/test-framework/core/src/main/java/org/keycloak/testframework/server/EmbeddedKeycloakServer.java
+++ b/test-framework/core/src/main/java/org/keycloak/testframework/server/EmbeddedKeycloakServer.java
@@ -1,11 +1,14 @@
 package org.keycloak.testframework.server;
 
+import java.util.Optional;
 import java.util.concurrent.TimeoutException;
 
 import org.keycloak.Keycloak;
 import org.keycloak.common.Version;
+import org.keycloak.it.utils.Maven;
 
 import io.quarkus.maven.dependency.Dependency;
+import org.eclipse.aether.artifact.Artifact;
 
 public class EmbeddedKeycloakServer implements KeycloakServer {
 
@@ -18,7 +21,10 @@ public class EmbeddedKeycloakServer implements KeycloakServer {
         this.tlsEnabled = tlsEnabled;
 
         for(Dependency dependency : keycloakServerConfigBuilder.toDependencies()) {
-            builder.addDependency(dependency.getGroupId(), dependency.getArtifactId(), "");
+            var version = Optional.ofNullable(Maven.getArtifact(dependency.getGroupId(), dependency.getArtifactId()))
+                    .map(Artifact::getVersion)
+                    .orElse("");
+            builder.addDependency(dependency.getGroupId(), dependency.getArtifactId(), version);
         }
 
         keycloak = builder.start(keycloakServerConfigBuilder.toArgs());

